includes/class-amp-theme-support.php

```diff
@@ -1036,10 +1055,36 @@ public static function prepare_response( $response, $args = array() ) {
 				'allow_dirty_styles'      => self::is_customize_preview_iframe(), // Dirty styles only needed when editing (e.g. for edit shortcodes).
 				'allow_dirty_scripts'     => is_customize_preview(), // Scripts are always needed to inject changeset UUID.
 				'disable_invalid_removal' => $is_validation_debug_mode,
+				'enable_response_caching' => ! defined( 'WP_DEBUG' ) || true !== WP_DEBUG || $is_validation_debug_mode,
 			),
 			$args
 		);
 
+		// Return cache if enabled and found.
+		if ( true === $args['enable_response_caching'] ) {
+			self::set_response_cache_key( array(
+				$args,
+				$response,
+				self::$sanitizer_classes,
+				self::$embed_handlers,
+				AMP__VERSION,
+			) );
+
+			if ( isset( $_REQUEST[ self::FLUSH_RESPONSE_CACHE_VAR ] ) ) { // WPCS: csrf ok.
+				wp_cache_delete( self::$response_cache_key, self::RESPONSE_CACHE_GROUP );
```

Comment:
Why does there need to be a way to manually flush the cache? Is this for development? If so, then there should be a cap check to make sure that only authorized users can delete the cache. Otherwise, I think this cache deletion ability could be removed.

Reply:
I agree, cap should be checked to flush cache. Yes it was to ease development as flushing cached may be be cumbersome depending on the setup and object caching mechanism used. Let's keep the code base clean though, I removed it [in this commit](https://github.com/Automattic/amp-wp/pull/1156/commits/9bcc4acfb28d87ba04b5f23c3793cce83b5deaa3).